### PR TITLE
fix: validate configuration inheritance for Endpoints and EndpointGro…

### PR DESCRIPTION
**Pull request by leleueri**
…up [APIM-607]

[APIM-607]: https://gravitee.atlassian.net/browse/APIM-607?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-gjtyzxqoux.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-607-endpoint-config-validation/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java [hunk 1]
```diff
@@ -81,6 +87,12 @@ public List<EndpointGroup> validateAndSanitize(List<EndpointGroup> endpointGroup
         return endpointGroups;
     }
 
+    private static void validateEndpointConfigInheritance(EndpointGroup endpointGroup, Endpoint endpoint) {
+        if (endpoint.isInheritConfiguration() && isNullOrEmpty(endpointGroup.getSharedConfiguration())) {
```
Comment (guillaumelamirand) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java:91]: Is it really necessary to force the user to set the value to false? Shouldn't we just ignore the "true" if no config is set at endpoint group level?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java [hunk 2]
```diff
@@ -59,6 +61,9 @@ public List<EndpointGroup> validateAndSanitize(List<EndpointGroup> endpointGroup
                 validateEndpointGroupType(endpointGroup.getType());
                 validateServices(endpointGroup.getServices());
                 validateEndpointsExistence(endpointGroup);
+
+                endpointService.validateConnectorConfiguration(endpointGroup.getType(), endpointGroup.getSharedConfiguration());
```
Comment (jhaeyaert) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java:65]: Having a closer look, I think something is still not good. The `validateConnectorConfiguration` method allows for a `null` value. It means that, if the endpoint configuration inheritance is disabled, it's still possible to have a `null` configuration right? Does it mean the gateway will throw an exception in that case? Why not adapt the http-proxy endpoint to protect itself against `null` configuration whatever it comes from the group or endpoint?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java (hunk 1 above)
Comment (leleueri) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java:91]: The inheritConfig flag is false by default, if the flag is set to true to me that mean the user do it with intention. So a null shared configuration is an error.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java (hunk 2 above)
Comment (leleueri) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java:65]: True,  `endpointService.validateConnectorConfiguration` ignore null value. Does it neam the config is required at both level EndpointGroup and endpoint whatever is the inherit value ?

I can add mechanism in the http-proxy (all proxy :thinking: ?) to have defensive behaviour but returning an error at API level is still valid to inform the API publisher that the definition is not valid IMHO.

**Comment by leleueri** [comment]
I will moved this PR to draft as the inheritance mechanism is challenged by the team.

**Comment by leleueri** [comment]
will be fixed by the split  of the EndpointCoinfiguration

**Pull request closed without merging**